File: #35.Search Insert Position/Search Insert Position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    var low=0;
    var high=nums.length-1;
    var mid;
    while(low<=high){
        mid=Math.floor((low+high)/2);
        if(nums[mid]==target){
            return mid;
        }else if(nums[mid]>target){
            high=mid-1;
        }else{
            low=mid+1;
        }
    }
    return low;
};

File: #35.Search Insert Position/Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int low=0,high=nums.size()-1;
        int mid;
        while(low<=high){
            mid=(low+high)/2;
            if(target==nums[mid])
                return mid;
            if(nums[mid]>target)
                high=mid-1;
            else
                low=mid+1;
        }
        return low;
    }
};